From 91ba5dac6d19dbffe513d994e569fdeacc12903a Mon Sep 17 00:00:00 2001
From: Kalyan Kondapally <kalyan.kondapally@intel.com>
Date: Sat, 3 Feb 2018 17:51:04 -0800
Subject: [PATCH 01/14] INTERNAL: i965: Relax permissions when we export prime
 handle for GEM.

This is needed in case we want to use mmap with dma-buf and
write into the buffer in CPU side. This is useful when
layers are rendered using software and we will either
need to scan or texture from these layers.
Check: https://patchwork.freedesktop.org/patch/56380/

Signed-off-by: Kalyan Kondapally <kalyan.kondapally@intel.com>
---
 src/mesa/drivers/dri/i965/brw_bufmgr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mesa/drivers/dri/i965/brw_bufmgr.c b/src/mesa/drivers/dri/i965/brw_bufmgr.c
index 536fe36e5e43..82032b18b033 100644
--- a/src/mesa/drivers/dri/i965/brw_bufmgr.c
+++ b/src/mesa/drivers/dri/i965/brw_bufmgr.c
@@ -1486,7 +1486,7 @@ brw_bo_gem_export_to_prime(struct brw_bo *bo, int *prime_fd)
    brw_bo_make_external(bo);
 
    if (drmPrimeHandleToFD(bufmgr->fd, bo->gem_handle,
-                          DRM_CLOEXEC, prime_fd) != 0)
+			  DRM_CLOEXEC | DRM_RDWR, prime_fd) != 0)
       return -errno;
 
    bo->reusable = false;
-- 
2.17.1

